src/lib/dao.js

```diff
@@ -572,6 +573,8 @@ foam.CLASS({
   methods: [
     function put(o) {
       this.a.push(o);
+    },
+    function eof() {
```

Comment:
This shouldn't be necessary, the Sink interface provides a default empty implementation of each method.